pkg/binding/controller.go

```diff
@@ -359,57 +360,36 @@ func verbsSupportInformers(verbs []string) bool {
 }
 
 // Event handler: enqueues the objects to be processed
-// At this time it is very simple, more complex processing might be required here
+// At this time it is very simple, more complex processing might be required
+// here.
 func (c *Controller) handleObject(obj any) {
-	rObj := obj.(runtime.Object)
-	c.logger.V(2).Info("Got object event", "obj", util.RefToRuntimeObj(rObj))
-	c.enqueueObject(obj, false)
+	c.logger.V(2).Info("Got object event", "obj", util.RefToRuntimeObj(obj.(runtime.Object)))
+	c.enqueueObject(obj)
 }
 
-// enqueueObject converts an object into a key struct which is then put onto the work queue.
-func (c *Controller) enqueueObject(obj interface{}, skipCheckIsDeleted bool) {
-	var key util.Key
-	var err error
-	if key, err = util.KeyForGroupVersionKindNamespaceName(obj); err != nil {
+// enqueueObject converts an object into an ObjectIdentifier struct which is
+// then put onto the work queue.
+func (c *Controller) enqueueObject(obj interface{}) {
+	objIdentifier, err := util.IdentifierForObject(obj.(util.MRObject), c.gvkToGvrMapper)
+	if err != nil {
 		utilruntime.HandleError(err)
 		return
 	}
-	if !skipCheckIsDeleted {
-		// we need to check if object was deleted. If deleted we need to enqueue the runtime.Object
-		// so we can still evaluate the label selectors and delete it from the clusters where it
-		// was deployed. This does not break the best practice of only storing the keys so that
-		// the latest version of the object is retrieved, as we know the object was deleted when
-		// we get a copy of the runtime.Object and there is no longer a copy on the API server.
-		_, err = c.getObjectFromKey(key)
-		if err != nil {
-			// The resource no longer exist, which means it has been deleted.
-			if errors.IsNotFound(err) {
-				deletedObj := copyObjectMetaAndType(obj.(runtime.Object))
-				// set deletion timestamp for this object to still be detected as being deleted
-				metaDeletedObj := deletedObj.(metav1.Object)
-				metaDeletedObj.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
-
-				key.DeletedObject = &deletedObj
 
-				c.workqueue.Add(key)
-			}
-			return
-		}
-	}
-	c.workqueue.Add(key)
+	c.workqueue.Add(objIdentifier)
```

Comment:
An Objectidentifier includes `ResourceVersion string`, which is not appropriate to include in an object reference in a workqueue.

Reply:
@MikeSpreitzer this was accidentally left here, I make no use of this field because I agree - it should not be here.

This is why I didn't want to continue working tonight 😢.

Feel free to remove it in the rebase PR.

Reply:
Removed.